src/cpp/session/SessionHttpMethods.cpp

```diff
@@ -768,12 +768,14 @@ void handleConnection(boost::shared_ptr<HttpConnection> ptrConnection,
 
             // exit status
             int status = switchToProject.empty() ? EXIT_SUCCESS : EX_CONTINUE;
-            if (options().getBoolOverlayOption(kLauncherSessionOption))
+            if (options().getBoolOverlayOption(kLauncherSessionOption) && (core::system::getenv("RSTUDIO_FORCE_NON_ZERO_EXIT_CODE") != "1"))
```

Comment:
Nit: we're already `using` core above, so we could shorten to `system::getenv()` here instead.